tools/serverpod_cli/test/generator/dart/server_code_generator/enum_field_test.dart

```diff
@@ -40,12 +39,19 @@ void main() {
     });
 
     test('then generated enum has static fromJson method', () {
-      expect(codeMap[expectedFileName],
-          contains('static Example? fromJson(int index)'));
+      expect(
+          codeMap[expectedFileName],
+          contains(config.serializeEnumValuesAsStrings
+              ? 'static Example? fromJson(String name)'
+              : 'static Example? fromJson(int index)'));
```

Comment:
Chore: Create separate groups for both enum types used in the test and only validate one in each group.

Depending on what the default value is in our GeneratorConfigBuilder we won't exercise one of these paths. We want to ensure that we support all paths possible to configure and should therefore have a test validating it.